src/ode_solver/ode_solver_factory.cpp

```diff
@@ -1,39 +1,28 @@
 #include "ode_solver_factory.h"
 #include "parameters/all_parameters.h"
 #include "ode_solver_base.h"
+//#include "runge_kutta_ode_solver.h"
 #include "explicit_ode_solver.h"
 #include "implicit_ode_solver.h"
 #include "rrk_explicit_ode_solver.h"
 #include "pod_galerkin_ode_solver.h"
 #include "pod_petrov_galerkin_ode_solver.h"
 #include <deal.II/distributed/solution_transfer.h>
+#include "runge_kutta_methods/runge_kutta_methods.h"
+#include "runge_kutta_methods/rk_tableau_base.h"
 
 namespace PHiLiP {
 namespace ODE {
 
 template <int dim, typename real, typename MeshType>
 std::shared_ptr<ODESolverBase<dim,real,MeshType>> ODESolverFactory<dim,real,MeshType>::create_ODESolver(std::shared_ptr< DGBase<dim,real,MeshType> > dg_input)
 {
-    std::cout << "Creating ODE Solver..." << std::endl;
     using ODEEnum = Parameters::ODESolverParam::ODESolverEnum;
-    ODEEnum ode_solver_type = dg_input->all_parameters->ode_solver_param.ode_solver_type;
-    if(ode_solver_type == ODEEnum::explicit_solver)        return std::make_shared<ExplicitODESolver<dim,real,MeshType>>(dg_input);
-    if(ode_solver_type == ODEEnum::implicit_solver)        return std::make_shared<ImplicitODESolver<dim,real,MeshType>>(dg_input);
-    if constexpr(dim==1){
-        //RRK is only implemented for Burgers on collocated nodes, 1D
-        const bool use_collocated_nodes = dg_input->all_parameters->use_collocated_nodes;
-        using PDEEnum = Parameters::AllParameters::PartialDifferentialEquation;
-        const PDEEnum pde_type = dg_input->all_parameters->pde_type;
-        const bool use_inviscid_burgers = (pde_type == PDEEnum::burgers_inviscid);
-        if ((ode_solver_type == ODEEnum::rrk_explicit_solver) && 
-                use_collocated_nodes && use_inviscid_burgers){
-            return std::make_shared<RRKExplicitODESolver<dim,real,MeshType>>(dg_input);
-        }
-        else{
-            display_error_ode_solver_factory(ode_solver_type, false);
-            return nullptr;
-        }
-    }
+    const ODEEnum ode_solver_type = dg_input->all_parameters->ode_solver_param.ode_solver_type;
+    if((ode_solver_type == ODEEnum::runge_kutta_solver)||(ode_solver_type == ODEEnum::rrk_explicit_solver))     
+        return create_RungeKuttaODESolver(dg_input);
+    if(ode_solver_type == ODEEnum::implicit_solver)         
+        return std::make_shared<ImplicitODESolver<dim,real,MeshType>>(dg_input);
```

Comment:
Add `pcout << "Creating ODE Solver..." << std::endl;` before returning the ImplicitODESolver; since it was removed at the top of the function

Reply:
That line was only at the top of `create_ODESolver(dg)`, not `create_ODESolver(dg,pod)` or `create_ODESolver_manual`, so I assumed it was left here from previous debugging, and removed it. The print statements from the RK solvers are intended to confirm the number of stages, rather than to confirm that the factory has been called. Is that okay? Otherwise I'll add print statements to all of the functions.

Reply:
Yeah add it to all the functions, would be good. Similar to how we have a print statement for Creating DG somewhere; for large runs without enough memory, its helpful for seeing where the code fails

Reply:
For sure, that's a good reason! I've just added `"Creating ODE Solver..." ` at the top of every `create_ODESolver` function. I think that's the simplest way & will require the least maintenance going forward - RK-type solvers now print two lines but I think that's fine.